File: estructura de datos/rectangulo.py
import math

class Rectangulo:
    def __init__(self, largo=0, ancho=0):
        self.__largo = largo
        self.__ancho = ancho
    
    def get__largo(self)->float:
        return self.__largo
    
    def get__ancho(self)->float:
        return self.__ancho
    
    def set_largo(self, largo:float)->None:
        self.__largo = largo
    
    def set_ancho(self, ancho:float)->None:
        self.__ancho = ancho
    
    def esRectangulo(self):
        return self.__largo>0 and self.__ancho>0
    
    def area(self):
        return self.__largo * self.__ancho
    
    def perimetro(self):
        return 2*(self.__largo + self.__ancho)
def main():
    try:
        largo=float(input("Ingrese el largo del rectangulo: "))
        ancho=float(input("Ingrese el ancho del rectangulo: "))
        r = Rectangulo(largo,ancho)

        if r.esRectangulo():
            print(f"el area del rectangulo: {r.area():.2f}")
            print(f"el perimetro del rectangulo: {r.perimetro():.2f}")
        else:
            print("no es un rectangulo")
    except ValueError:
        print("No hay sistema, por favor ingrese un numero que sea valido")
if __name__ == "__main__":
    main()

File: estructura de datos/FiguraGeometrica/src/figurageometrica/rectangulo.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package figurageometrica;

/**
 *
 * @author Javier
 */
public class rectangulo {
    private double largo, ancho;

    public double getLargo() {
        return largo;
    }

    public void setLargo(double largo) {
        this.largo = largo;
    }
    public rectangulo(){
        this.largo =0;
        this.ancho=0;
}

    public double getAncho() {
        return ancho;
    }

    public void setAncho(double ancho) {
        this.ancho = ancho;
    }

    public rectangulo(double largo, double ancho) {
        this.largo = largo;
        this.ancho = ancho;
    }
    public boolean esRectangulo(){
        return largo>0 && ancho>0;
    }
    public double area(){
        return largo * ancho;
    }
    public double perimetro(){
        return 2*(largo+ancho);
    }
}
